colossalai/zero/sharded_optim/sharded_optim_v3.py

```diff
@@ -0,0 +1,155 @@
+from enum import Enum
+from typing import Dict, Optional, Union
+
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from colossalai.amp.naive_amp._fp16_optimizer import DynamicGradScaler
+from colossalai.context.parallel_mode import ParallelMode
+from colossalai.core import global_context as gpc
+from colossalai.nn.optimizer import ColossalaiOptimizer
+from colossalai.zero.sharded_model import ShardedModelV2
+from torch import Tensor
+from torch.distributed import ProcessGroup
+from torch.nn.parameter import Parameter
+from torch.optim import Optimizer
+
+from ..shard_utils import TensorShardStrategy
+from ..sharded_param import ShardedTensor
+from ._utils import has_inf_or_nan
+
+
+class OptimState(Enum):
+    SCALED = 1
+    UNSCALED = 2
+
+
+class ShardedOptimizerV2(ColossalaiOptimizer):
+
+    def __init__(self,
+                 optimizer: Optimizer,
+                 sharded_model: Union[nn.Module, ShardedModelV2],
+                 cpu_offload: bool = False,
+                 initial_scale: float = 2**32,
+                 min_scale: float = 1,
+                 growth_factor: float = 2,
+                 backoff_factor: float = 0.5,
+                 growth_interval: float = 1000,
+                 hysteresis: float = 2,
+                 max_scale: int = 2**32,
+                 dp_process_group: Optional[ProcessGroup] = None,
+                 mp_process_group: Optional[ProcessGroup] = None) -> None:
+        super().__init__(optimizer)
+        self.model: Union[nn.Module, ShardedModelV2] = sharded_model
```

Comment:
I just wonder if we can do it in this way.
If the class receive a Module type sharded_model. It just converts it into a ShardedModelV2

sharded_model = ShardedModelV2(sharded_model， ..)